lib/kitchen/driver/gce.rb

```diff
@@ -141,26 +141,97 @@ def destroy(state)
         state.delete(:zone)
       end
 
+      def old_disk_configuration_present?
+        !config[:autodelete_disk].nil? || !config[:disk_size].nil? || !config[:disk_type].nil?
+      end
+
+      def new_disk_configuration_present?
+        !config[:disks].nil?
+      end
+
+      def create_disks_config
+        # This can't be present in default_config because we couldn't
+        # determine which disk configuration the user used otherwise
+        disk_default_config = {
+          autodelete_disk: true,
+          disk_size: 10,
+          disk_type: "pd-standard",
+        }
+
+        if old_disk_configuration_present?
+          # If the old disk configuration is used,
+          # we'll convert it to the new one
+          config[:disks] = {}
+          config[:disks][server_name] = {
```

Comment:
The use of `server_name` here results in an error `undefined local variable or method 'server_name'`. This should probably be `:disk1` instead.

It's a bit confusing, but `server_name` in this class is not an instance method. Everywhere it is used, `server_name` is a local variable for that method. The value is generated once and stored in `state[:server_name]` and from there it is retrieved and set again for other methods that need to know the name.

I don't think this `create_disks_config` method needs to use the server name, though. Using `:disk1` here will work, much like the final case in this method where there is no explicit disk configuration, so a disk config is created named `disk1`.

Reply:
The current situation has `server_name` passed in to the `boot_disk()` method from [`create_instance_object`](https://github.com/test-kitchen/kitchen-google/blob/master/lib/kitchen/driver/gce.rb#L330) which itself had `server_name` passed in from the [`create`](https://github.com/test-kitchen/kitchen-google/blob/master/lib/kitchen/driver/gce.rb#L101) method where the name was [generated and stored as a local variable](https://github.com/test-kitchen/kitchen-google/blob/master/lib/kitchen/driver/gce.rb#L98). It's not _clean_, by any means.

The server name is not necessary here where the disk configuration is being built. It only needs a sensible short name for the disk, similar to the ones written in the README that we expect a human to write into their kitchen.yml. With your new `create_disks` method, [the server name and the disk name are concatenated](https://github.com/test-kitchen/kitchen-google/pull/62/files#diff-6cfd6bc13548320b2c96b01086644ddcR431) for the name used for the "real" disk created in GCE. So giving this disk a short, sensible default name like "disk1" will work. (It did when I tested [the simplified transform from old to new in one of my other comments](https://github.com/test-kitchen/kitchen-google/pull/62#discussion_r183055011).)